src/core/symbology/qgsfillsymbollayer.cpp

```diff
@@ -3909,3 +3912,306 @@ void QgsRasterFillSymbolLayer::applyPattern( QBrush &brush, const QString &image
 
   brush.setTextureImage( img );
 }
+
+
+//
+// QgsRandomMarkerFillSymbolLayer
+//
+
+QgsRandomMarkerFillSymbolLayer::QgsRandomMarkerFillSymbolLayer( int pointCount, unsigned long seed )
+  : mPointCount( pointCount )
+  , mSeed( seed )
+{
+  setSubSymbol( new QgsMarkerSymbol() );
+}
+
+QgsSymbolLayer *QgsRandomMarkerFillSymbolLayer::create( const QgsStringMap &properties )
+{
+  const int pointCount = properties.value( QStringLiteral( "point_count" ), QStringLiteral( "1" ) ).toInt();
+
+  unsigned long seed = 0;
+  if ( properties.contains( QStringLiteral( "seed" ) ) )
+    seed = properties.value( QStringLiteral( "seed" ), QStringLiteral( "0" ) ).toULong();
```

Comment:
If the computation of a random seed number (currently in the else clause) isn't too expensive, what about always coming up which such a number and use that instead of "0" as fallback value here?

Reply:
Actually this default isn't being used -- the .contains() check above negates it

Reply:
@nyalldawson , ah true, an dif the seed value isn't a number (i.e. somebody setting a wrong property value in the map), it wouldn't fallback to the default anyways. All good, sorry for the noise ;-P